src/tools/ceph_objectstore_tool.cc

```diff
@@ -861,14 +861,24 @@ int get_attrs(
 	  cout << "snapset " << snapset << std::endl;
 	  if (!snapset.is_legacy()) {
 	    for (auto& p : snapset.clone_snaps) {
-	      hobject_t clone = hoid.hobj;
-	      clone.snap = p.first;
+	      ghobject_t clone = hoid;
+	      clone.hobj.snap = p.first;
 	      set<snapid_t> snaps(p.second.begin(), p.second.end());
+	      if (!store->exists(coll, clone)) {
```

Comment:
241f8ad6e2a7df739e2154ea41a1884c78502166 has the change you can cherry-pick here.